src/libstd/sys/unix/net.rs

```diff
@@ -140,10 +140,7 @@ impl Socket {
             }
 
             let timeout = timeout - elapsed;
-            let mut timeout = timeout
-                .as_secs()
-                .saturating_mul(1_000)
-                .saturating_add(timeout.subsec_nanos() as u64 / 1_000_000);
+            let mut timeout = timeout.as_millis() as u64;
```

Comment:
This does not maintain the saturating behavior of the original implementation.

Reply:
So, this approach isn't correct? If so, we should just close that issue, I think.

Reply:
The conversion should saturate to `u64::max_value()` if it would be unrepresentable.

Reply:
Hm, the casting should saturate `as_millis()` value. Should it be saturated in stages like the current behavior?

Reply:
`as` casts do not saturate, they just discard the high bits.

Reply:
> If so, we should just close that issue, I think.

Then I think we should just close this PR and that issue as above my comment, right?

Reply:
Ah, sorry, I just got what you mean, will fix...

Reply:
Seems reasonable.

Reply:
Okay, closing.